### PR TITLE
Fix EZP-29116 : Conversions from ezxmltext to ezrichtext fails if ezxmtext have duplicate xhtml ids

### PR DESCRIPTION
**Pull request by vidarl**
This fix is dependent on https://github.com/ezsystems/ezpublish-kernel/pull/2312 which actually deals with the xml conversion..

The change in this PR 
- outputs warnings of duplicated ids are detected and new ones had to be generated due to that.
- Refactors so that it is possible to make tests for ezxmltext -> richtext conversion
- Adds tests for duplicate ids

### CODE REVIEW

**Comment by vidarl** [comment]
The tests will fail until https://github.com/ezsystems/ezpublish-kernel/pull/2312 is merged.

**Comment by vidarl** [comment]
@adamwojs : Could you have a look on this one too? I was not able to add you as reviewer

File: tests/lib/FieldType/Converter/RichTextTest.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * This file is part of the eZ Platform XmlText Field Type package.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ *
+ * @version //autogentag//
```
Comment (adamwojs) [inline tests/lib/FieldType/Converter/RichTextTest.php]: Nitpick: `@version` is not needed anymore.

File: lib/FieldType/XmlText/Converter/RichText.php [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        );
+
+        $this->validator = new Validator(
+            array(
+                './vendor/ezsystems/ezpublish-kernel/eZ/Publish/Core/FieldType/RichText/Resources/schemas/docbook/ezpublish.rng',
+                './vendor/ezsystems/ezpublish-kernel/eZ/Publish/Core/FieldType/RichText/Resources/schemas/docbook/docbook.iso.sch.xsl',
+            )
+        );
+    }
+    function removeComments(DOMDocument $document)
```
Comment (adamwojs) [inline lib/FieldType/XmlText/Converter/RichText.php]: Nitpick: Missing new line between methods. Can we also specify the visibility of the method?

Comment (vidarl) [reply lib/FieldType/XmlText/Converter/RichText.php]: Yeah.. both nitpicks are a result of cut&paste... I'll fix

Comment (vidarl) [reply lib/FieldType/XmlText/Converter/RichText.php]: fixed

File: bundle/Command/ConvertXmlTextToRichTextCommand.php [hunk 3]
```diff
@@ -174,8 +141,9 @@ function convertFieldDefinitions($dryRun, OutputInterface $output)
         $output->writeln("Converted $count ezxmltext field definitions to ezrichtext");
     }
 
-    function convertFields($dryRun, $contentObjectId, OutputInterface $output)
+    function convertFields($dryRun, $contentObjectId, $checkDuplicateIds, OutputInterface $output)
```
Comment (adamwojs) [inline bundle/Command/ConvertXmlTextToRichTextCommand.php]: Nitpick: missing visibility specification for method.

File: lib/FieldType/XmlText/Converter/RichText.php [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    public function __construct(LoggerInterface $logger = null)
+    {
+        $this->logger = $logger;
+
+        $this->converter = new Aggregate(
+            array(
+                new ToRichTextPreNormalize(new Expanding(), new EmbedLinking()),
+                new Xslt(
+                    './vendor/ezsystems/ezpublish-kernel/eZ/Publish/Core/FieldType/RichText/Resources/stylesheets/ezxml/docbook/docbook.xsl',
```
Comment (adamwojs) [inline lib/FieldType/XmlText/Converter/RichText.php]: Specifying path relative to `__DIR__` is a bit safer for me, especially when it was moved from the commend

Comment (adamwojs) [reply lib/FieldType/XmlText/Converter/RichText.php]: > So, I could add some logic and check whatever vendor/ezsystems/ezpublish-kernel or ../ezpublish-kernel exists, and set the path accordingly, relative to __DIR__

@vidarl No, in this circumstances let's leave it as is :wink:

File: bundle/Command/ConvertXmlTextToRichTextCommand.php (hunk 3 above)
Comment (vidarl) [reply bundle/Command/ConvertXmlTextToRichTextCommand.php]: Fixed

**Pull request merged**